sources/packages/backend/apps/api/src/auth/_tests_/e2e/maintenance-mode.guard.e2e-spec.ts

```diff
@@ -0,0 +1,266 @@
+import { Test, TestingModule } from "@nestjs/testing";
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { KeycloakConfig } from "@sims/auth/config";
+import { createE2EDataSources, E2EDataSources } from "@sims/test-utils";
+import {
+  createZeebeModuleMock,
+  ConfigServiceMockHelper,
+} from "@sims/test-utils/mocks";
+import { AppModule } from "../../../app.module";
+import { DiscoveryModule } from "@golevelup/nestjs-discovery";
+import { DataSource } from "typeorm";
+import {
+  BEARER_AUTH_TYPE,
+  FakeStudentUsersTypes,
+  getAESTToken,
+  getAuthorizedLocation,
+  getExternalUserToken,
+  getInstitutionToken,
+  getStudentToken,
+  InstitutionTokenTypes,
+} from "../../../testHelpers";
+import { AuthTestController } from "../../../testHelpers/controllers/auth-test/auth-test.controller";
+import { InstitutionUserTypes } from "@sims/sims-db";
+
+describe("MaintenanceModeGuard (e2e)", () => {
+  // Nest application to be shared for all e2e tests
+  // that need execute a HTTP request.
+  let app: INestApplication;
+  // Token to be used for AEST e2e tests that need test
+  // the authentication endpoints.
+  // This token is retrieved from KeyCloak.
+  let moduleFixture: TestingModule;
+  let db: E2EDataSources;
+  let configServiceMockHelper: ConfigServiceMockHelper;
+
+  beforeAll(async () => {
+    await KeycloakConfig.load();
+    moduleFixture = await Test.createTestingModule({
+      imports: [AppModule, createZeebeModuleMock(), DiscoveryModule],
+      // AuthTestController is used only for e2e tests and could be
+      // changed as needed to implement more test scenarios.
+      controllers: [AuthTestController],
+    }).compile();
+    app = moduleFixture.createNestApplication();
+    app.init();
+    const dataSource = moduleFixture.get(DataSource);
+    db = createE2EDataSources(dataSource);
+    configServiceMockHelper = new ConfigServiceMockHelper(app);
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
```

Comment:
For other tests, we have been adding the `afterAll` method to the end of the file.